docs/rfc/active/graph_ql_frontend/README.md

```diff
@@ -0,0 +1,617 @@
+
+To support traversal of this relationship, we additionally include a ``CorpUser`` type:
+
+```graphql
+"""
+Represents the CorpUser GMA Entity
+"""
+type CorpUser {
+
+    urn: String!
+
+    username: String!
+
+    info: CorpUserInfo
+
+    editableInfo: CorpUserEditableInfo
+}
+```
+
+#### 2. **Extend the 'Query' type**: 
+
+GraphQL defines a top-level 'Query' type that serves as the entry point for reads against the graph. This is extended to support querying the new entity type.
+
+```graphql
+type Query {
+    dataset(urn: String!): Dataset # Add this! 
+    datasets(urn: [String]!): [Dataset] # Or if batch support required, add this! 
+}
+```
+
+#### 3. **Define & Register DataLoaders**
+This is illustrated in the previous section. It involves extending `DataLoader` and registering the loader in the `DataLoaderRegistry`.
+
+#### 4. **Define & Register DataFetcher (Data Resolver)**
+This is illustrated in the previous section. It involves implementing the DataFetcher interface and attaching it to fields in the graph.
+
+#### 5. **Query your new entity**
+Deploy & start issuing queries against the graph. 
+
+```graphql
+// Input
+query datasets($urn: String!) { 
+    dataset(urn: $urn) { 
+        urn 
+        ownership {
+            owners {
+                owner {
+                    username
+                }
+            }
+        }
+    }
+}
+
+// Output 
+{
+    "data": {
+        "dataset": {
+            "urn": "urn:li:dataset:(urn:li:dataPlatform:foo,bar,PROD)",
+            "ownership": {
+                "owners": [
+                    {
+                        "owner": {
+                            "username": "james joyce",
+                        }
+                    }
+                ]
+            }
+        }
+    }
+}
+```
+
+## How we teach this
+
+We will create user guides that cover: 
+- Modeling & onboarding entities to the GQL type system
+- Modeling & onboarding entity aspects to the GQL type system
+- Modeling & onboarding relationships among entities to the GQL type system 
+
+## Alternatives
+Keep the resource-oriented approach, with different endpoints for each entity / aspect. 
+
+## Rollout / Adoption Strategy
+
+1. Rollout CRUD-style reads against entities / aspects in the graph
+      - Entities: Dataset, CorpUser, DataPlatform
+      - Relationships: Dataset->CorpUser, Dataset->DataPlatform, [if there is demand] CorpUser -> Dataset
+2. Rollout CRUD-style writes against entities / aspects in the graph
+3. Rollout full text search against entities / aspects in the graph
+4. Rollout browse against entities / aspects in the graph
+5. Migrate client-side apps to leverage new GraphQL API 
+    - Build out parallel data fecthing layer, where the existing models are populated either by the old clients or the new GQL clients. Place this behind a client-side feature flag. 
+    - Allow users of DataHub to configure which API they want to run with, swap at their own pace.
+    
+## Unresolved questions
```

Comment:
Can you add something here about resolving PDL and GQL?  I'm not a big fan of the added maintenance of adding yet another schema file. Granted, maybe it is needed if the GQL ends up being a different model than the PDL ones, but if they're the same some converter would be nice.

Reply:
This has been addressed in a future revision through a disclaimer about the assumption of auto-generation of GQL from PDL.